### PR TITLE
refactor: fix type hints of train_test_split

### PR DESCRIPTION
**Pull request by baggiponte**
## What does this implement/fix? Explain your changes.

Fix the type annotations of train_test_split.

**Note**. Technically is a breaking change since I made `eager` a keyword only argument. If we plan to release in 0.10 we should mark it as such. Though I would assume that no one with benign intentions calls `train_test_split(4, true)`.

### CODE REVIEW

File: functime/cross_validation.py [hunk 1]
```diff
@@ -18,12 +27,37 @@ def train_test_split(
 
     Returns
     -------
-    splitter : Callable[pl.LazyFrame, Tuple[pl.LazyFrame, pl.LazyFrame]]
-        Function that takes a panel LazyFrame and returns tuple of train / test LazyFrames.
+    splitter : Callable[pl.LazyFrame | pl.DataFrame, Tuple[pl.LazyFrame, pl.LazyFrame]]
```
Comment (FBruzzesi) [inline functime/cross_validation.py:30]: Shouldn't this coincide with above annotation? Namely
 ```suggestion
     splitter : Callable[[pl.LazyFrame | pl.DataFrame], Tuple[pl.LazyFrame, pl.LazyFrame] | Tuple[pl.DataFrame, pl.DataFrame]]
```

**Review by FBruzzesi — COMMENTED** [review]
Following @baggiponte [comment](https://github.com/functime-org/functime/pull/168#issuecomment-1884242541), I took a quick look at this and left a comment for the return type in function docstring.

File: functime/cross_validation.py (hunk 1 above)
Comment (baggiponte) [reply functime/cross_validation.py:30]: thanks for spotting that!

**Comment by FBruzzesi** [comment]
> Following @baggiponte [comment](https://github.com/functime-org/functime/pull/168#issuecomment-1884242541), I took a quick look at this and left a comment for the return type in function docstring.

Actually let me add some additional comments since I am here 😂

- The docstring for `train_test_split` says: """Return a time-ordered train set and test set given `test_size`.""" which is not accurate as it returns the callable.
- As the callable is returned, I would add proper docstring to that as well so that one can inspect it if needed

**Comment by baggiponte** [comment]
> > Following @baggiponte [comment](https://github.com/functime-org/functime/pull/168#issuecomment-1884242541), I took a quick look at this and left a comment for the return type in function docstring.
> 
> Actually let me add some additional comments since I am here 😂
> 
> * The docstring for `train_test_split` says: """Return a time-ordered train set and test set given `test_size`.""" which is not accurate as it returns the callable.
> * As the callable is returned, I would add proper docstring to that as well so that one can inspect it if needed

Thanks.

I was wondering whether we should remove the `eager` arg anyway and have a simpler signature `def train_test_split[DF: (pl.DataFrame, pl.LazyFrame)](data: DF, ...) -> Callable[[DF], tuple[DF, DF]]: ...`. No overloads to write.

**Comment by baggiponte** [comment]
@FBruzzesi I should merge this before you go on with #168. I wanted to discuss with other maintainers @ngriffiths13 whether we can get rid of the `eager` argument. The reasoning is: if we pass a DataFrame, we return a DataFrame. Same with LazyFrame. In this way, we remove code we have to annotate and test. The user can simply call `lazy()` or `collect()` before splitting, if they want.
